docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+I.e. prevent destruction of restored spoke clusters.
+
+##### Suggested solution for automating the flow (make it transparent to the user):
+
+For BMHs in the active hub: (In assisted-service -> `bmh_agent_controller`)
+* Once the BMH is in 'Provisioned' state:
+  * Extract BMH's status and add `baremetalhost.metal3.io/status` to the object metadata.
+  * Add also, add the `baremetalhost.metal3.io/paused` annotation:
+    * This is required since once the annotation is added, BMO updates the BMH's status and delete the annotation.
+    * The [paused](https://github.com/metal3-io/baremetal-operator/blob/main/docs/api.md#pausing-reconciliation) annotation, is similar to the `detached` annotation we're already using - but, it's also pausing the reconcile loop. Thus, keeping the status annotation intact.
+
+For BMHs in the passive hub: (In assisted-service -> `bmh_agent_controller`)
+* When the BMH is applied, the status will be missing - since reconcile is paused.
+  * To restore the original status:
+    * Remove the `baremetalhost.metal3.io/paused` annotation
+      * Status is restored, but `baremetalhost.metal3.io/status` annotation is removed.
+    * Hence, add the status and paused annotation back
+      * By detecting a BMH with `provisioned` state and a missing status annotation.  
+
+#### Solution for deleted Agent CRs
+
+1. Don't delete the Agent CR when an associated host doesn't exits on DB.
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * Some of the inventory data can be restored from annotations.
+     E.g.:
+     * inventory.agent-install.openshift.io/cpu-architecture: x86_64
+     * inventory.agent-install.openshift.io/cpu-virtenabled: "true"
+     * inventory.agent-install.openshift.io/host-isvirtual: "true"
+     * inventory.agent-install.openshift.io/host-manufacturer: RedHat
+     * inventory.agent-install.openshift.io/host-productname: KVM
+     * inventory.agent-install.openshift.io/storage-hasnonrotationaldisk: "false"
+   * Note: We might need to add more annotations for additional inventory data.
+   
+### Risks and Mitigations
+
+* In general, these changes might affect flows on spoke cluster (before and after restore), need to ensure no regressions on all scenarios.
+* Adding the 'paused' annotation means a stale BMH status, need to check if it's crucial.
```

Comment:
Also what if we want to deprovision using BMH - will this pause annotation affect that?

Reply:
We already have the detached annotation which our controller removes when we need to deprovision (unbind or delete). I imagine we'll do the same for paused. Especially if it fully replaces detached.

Reply:
Sweet yeah just want to make sure it's called out that's something in our scope too!

Reply:
We do have some people complaining about the BMH status not being accurate.

Do we want to make this entire procedure (pausing BMH and saving status annotation) opt-in? Maybe it would be good to put this behind a feature flag in any case to start?

Reply:
> We do have some people complaining about the BMH status not being accurate.
> 
> Do we want to make this entire procedure (pausing BMH and saving status annotation) opt-in? Maybe it would be good to put this behind a feature flag in any case to start?

Good point. As an alternative, it could be opt-out perhaps? I mean, disabling this logic when required.
I.e. Not sure how crucial it is to keep the BMH status updated once already provisioned:/

@avishayt @gamli75 @oourfali WDYT?

Reply:
As long as we have some flag I'm fine.

I'm not as concerned with it being enabled or not.